### DISCUSSION TITLE
Setting a durable?

### DISCUSSION THREAD

**Discussion by joshuaziering** [postmanlabs/postman-flows#236] [Q&A] [answered]
I have an endpoint that returns a Hash that I need to use in subsequent request. I think the Durable is the right element to use. The hash is in a formatted JSON response body. Why can't I access that response body in the durables element? Here's my setup:
<img width="791" alt="Screen Shot 2022-01-31 at 4 13 59 PM" src="https://user-images.githubusercontent.com/1530131/151894121-a1639477-01c8-4336-9ebf-ea669bd103b6.png">

I have the Parse Body set to auto in the first endpoint. What am I missing? I dont even see an option to interact with the response body.

<img width="763" alt="Screen Shot 2022-01-31 at 4 18 39 PM" src="https://user-images.githubusercontent.com/1530131/151894273-1beade02-ab8c-4642-809d-72cf9a59eb50.png">

What am I missig?

**Comment by tomalazarov** [comment, marked as answer]
I guess what's happening is that you haven't saved an example of the response of the request. You see that exclamation mark on the "Check to get a Iaanc" request? That means you didn't saved an example. Navigate to the request itself and execute it in order to get the proper, expected and correct response body. Once you do that the exclamation mark will be gone and now the next attached block will know how the response body should look like and you'll be able to select data Record > body > etc. desired value.
<img width="1047" alt="Screenshot 2022-02-01 at 10 53 56" src="https://user-images.githubusercontent.com/48013428/151939883-0cc1f78c-8236-476f-b455-079669d087bb.png">

**Reply by xbranko** [reply]
What should one do when the request returns XML? In the (not flows) request tab _Tests_ one can do arbitrary JavaScript, in my case I do xml2json transformation. What is the equivalent action in flows? In the _Assign values to variables_ window, there are two options in the drop down menu for  **Parse Body**: _Auto_ and _Json_.  When I select the variable, I only see the generic options (like in the picture in the question). I have saved the example for the request, as you suggested, so the exclamation mark in yellow triangle is gone.

**Reply by saswatds** [reply]
Hey @xbranko we have not yet added xml support yet for flows. XML is indeed on the most used serialization formats after JSON, therefore I will add it to the core so that you don't have to worry about converting.

**Reply by DiscoMcDisco** [reply]
I've confirmed with my devs that the response we get is JSON, I've saved my 200 OK response as an example (so no warning symbol) and I still see Body *string* (you'll see I tried to do it manually lol) - do i have to save the example with the right pretty formatting perhaps?

<img width="1332" alt="Screenshot 2022-03-07 at 16 48 52" src="https://user-images.githubusercontent.com/5165228/157079568-f23afe04-3ca0-4df2-8c59-d34ef21e2e6d.png">

<img width="789" alt="Screenshot 2022-03-07 at 16 15 00" src="https://user-images.githubusercontent.com/5165228/157078296-dc906be0-94e0-45e1-9553-4e669c30d4be.png">

Desktop App is fully up to date (as far as I can tell, did it about 30 mins ago to retry and still fails).

**Reply by DiscoMcDisco** [reply]
@tomalazarov / @saswatds : sorry, know it’s cheeky but it’s been a couple days and I just can’t get this working :(

**Reply by saswatds** [reply]
Hey @DiscoMcDisco Can you confirm that the language selection on the Examples Tab is set to JSON? 
![image](https://user-images.githubusercontent.com/8796447/157428903-23338588-91ab-4f51-9c9e-9e7bf1f75f59.png)

**Reply by DiscoMcDisco** [reply]
correct yes :)

<img width="1371" alt="Screenshot 2022-03-09 at 11 02 59" src="https://user-images.githubusercontent.com/5165228/157429468-886bb9cf-0092-4600-99c8-3103d313f966.png">

**Reply by saswatds** [reply]
This is very strange. Are there any other examples present for the same request? These are the only two problematics vectors for data type inference to fail. Please reach out to help@postman.com if the problem persists, our support team can connect to further debug this issue.

**Reply by DiscoMcDisco** [reply]
thanks for the advice matey, i'll reach out :)

**Comment by sunilvp** [comment]
Hi , 
If for same request the response fields might change. In that use case if i save both the output with "save as example", i am not getting option to select many of the fields from the body.

**Reply by saswatds** [reply]
Hey @sunilvp. Yeah, currently, this is a limitation with flows. It cannot identify different examples. It just uses the first example that it finds with a 200 status code. We are working to improve the type system to work with all these different discriminated types. 

Until we figure that out, if possible, you could define the different variants of the same request as separate requests.

**Reply by sunilvp** [reply]
Got it thanks a lot.

**Comment by JesusVFR** [comment]
Hi all,

can you help on my scenario:

1) SentRequest -> Login , we get back the bearer JWT token ad saved as **durable variable**
![image](https://user-images.githubusercontent.com/75691077/158053731-fd47619a-acc6-4f93-b7df-b108357130d3.png)

2) I want to call another "SendRequest" block, being an authenticated service, i need to pass the **bearer token** on headers, setting the **Authorization** key  how can I do that ? 
![image](https://user-images.githubusercontent.com/75691077/158053906-a0b172a4-d0d3-4817-829b-7c83462fb1ba.png)

**Comment by DiscoMcDisco** [comment]
Nope, I don’t often use examples and flows is making me aware I need both them and possibly to shift to environments variables.

This call is one of very few with a saved valid example.

Note: when the example is saved it defaults to auto, which looks to be text. It’s set to JSON though now, once saved and saved again

Russ Watson-Thomas
07967 180 603
